src/MaaCore/Task/Miscellaneous/StartGameTaskPlugin.cpp

```diff
@@ -9,48 +9,53 @@ bool StartGameTaskPlugin::_run()
     if (m_client_type.empty()) {
         return false;
     }
+    // check for MAC / iOS
+    if (ctrler()->get_controller_type() == ControllerType::MacPlayTools) {
+        return ctrler()->start_game(m_client_type);
+    }
+    else {
+        // check for android version, because it leads to different magic values
+        // https://github.com/MaaAssistantArknights/MaaAssistantArknights/pull/8966#issuecomment-2094369694
+        if (ctrler()->get_version() > 8) {
+            do {
+                if (!ctrler()->start_game(m_client_type)) {
+                    return false;
+                }
 
-    // check for android version, because it leads to different magic values
-    // https://github.com/MaaAssistantArknights/MaaAssistantArknights/pull/8966#issuecomment-2094369694
-    if (ctrler()->get_version() > 8) {
-        do {
-            if (!ctrler()->start_game(m_client_type)) {
-                return false;
-            }
+                // https://github.com/MaaAssistantArknights/MaaAssistantArknights/pull/8961#issue-2277568882
+                // 167: magic value needs to be >164 but <172 (as that's max)
+                if (ctrler()->get_pipe_data_size() > 167) {
+                    return true;
+                }
 
-            // https://github.com/MaaAssistantArknights/MaaAssistantArknights/pull/8961#issue-2277568882
-            // 167: magic value needs to be >164 but <172 (as that's max)
-            if (ctrler()->get_pipe_data_size() > 167) {
-                return true;
-            }
+                sleep(1500);
 
-            sleep(1500);
+            } while (true);
```

Comment:
I think it's better to add a maximum number of retries.

Reply:
No idea what a good number would be.
Perhaps 50 just to make completely sure.